### ISSUE TITLE
High training loss fix

### ISSUE THREAD

**Issue by madhushankara** [wickaknight/LLM-fine-tune-with-doc#1]
===== TOKEN-LEVEL ANALYSIS =====
Loading model for token analysis...

Overall loss: 2.7551

Tokens with highest loss:
  'Ġledge': inf
  'Ġplaint': 16.6406
  'Ġshr': 14.3359
  'Ġamused': 13.8047
  'ĠGloss': 12.3906
  'Ġnumber': 10.9375
  'Ġcleaning': 10.3125
  'Ġdevour': 10.1641
  'Ġprecip': 10.0938
  'Ġtrot': 9.8516
